R/page.R

```diff
@@ -95,3 +95,15 @@ page_offset <- function(client, req, limit = Inf) {
   }
   return(agg_response)
 }
+
+optional_progress_bar <- function(...) {
+  if (requireNamespace("progress", quietly = TRUE)) {
+    progress::progress_bar$new(...)
+  } else {
+    # Return a mock object that behaves enough like a progress bar object
+    list(
+      tick = function() {},
+      finished = TRUE
+    )
```

Comment:
What happens without that `finished = TRUE`? I see it referenced at Would https://github.com/rstudio/connectapi/blob/443a2a2049df069a3606459e352569d01e25f06d/R/get.R#L294-L299 But maybe that could be changed / pulled in here and then `finsihed` isn't needed at all?

Reply:
I'll take a look at refactoring

Reply:
OOPH sorry about the copy pasta disfluencies there. I had two comments about that chunk elsewhere edited them into one and only did a half good job at that 🤦